client/src/http_default.rs

```diff
@@ -39,50 +43,52 @@ impl HttpReqExt for http::request::Builder {
     }
 }
 
-pub fn post<U, P, K, V>(
-    url: U,
-    body: Bytes,
-    query_params: P,
-    headers: Headers,
-) -> Result<Response<Bytes>>
+/// Default request builder & sender implemented on top of `attohttpc` crate.
+pub struct DefaultRequestBuilder<T: Body = attohttpc::body::Bytes<Bytes>>(
+    AttoHttpRequestBuilder<T>,
+);
+
+impl<T> DefaultRequestBuilder<T>
 where
-    U: AsRef<str>,
-    P: IntoIterator,
-    P::Item: Borrow<(K, V)>,
-    K: AsRef<str>,
-    V: ToString,
+    T: Body,
 {
-    let url = url.as_ref();
-    let response = attohttpc::post(url)
-        .bytes(body)
-        .params(query_params)
-        .set_headers(headers)?
-        .send()
-        .wrap_err_with(|| format!("Failed to send http post request to {}", url))?;
-    ClientResponse(response).try_into()
+    pub fn send(self) -> Result<Response<Bytes>> {
+        let Self(mut builder) = self;
```

Comment:
Why not just `let mut builder = self.0`? Or use `self.0` directly?

Reply:
Just a matter of personal taste. Tried to make it cleaner.

Reply:
@Arjentix this is also mentioned in our contributing guide.